### PR TITLE
[Merged by Bors] - chore: bump std

### PR DESCRIPTION
**Pull request by kim-em**
Some deleted lemmas have been upstreamed to Std.

Note that the statements of `List.zipWith_map_left` (and `_right`) have been changed, requiring slight changes here.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Lists.lean
Comment (semorrison) [inline Mathlib/SetTheory/Lists.lean:442]: @alexjbest do you undestand what is happening here?

(Let's not wait for a resolution to this before merging this PR.)

Comment (alexjbest) [reply Mathlib/SetTheory/Lists.lean]: I think its simply that when you make a mutual `def/theorem/instance` lean creates the internal autogenerated mutual parts as `def/theorem/instance` correspondingly, which seems non-ideal in the `instance` cases but not high priority to change as mutual instances are quite rare.

**Comment by alexjbest** [comment]
maintainer merge

File: Mathlib/SetTheory/Lists.lean
Comment (eric-wieser) [inline Mathlib/SetTheory/Lists.lean:441]:
```suggestion
-- This is an autogenerated declaration, so there's nothing we can do about it.
```
I don't think the comment needs to explain what `nolint` means.

File: Mathlib/Data/List/Basic.lean
Comment (eric-wieser) [inline Mathlib/Data/List/Basic.lean:1121]: I assume these were upstreamed; please mention that (maybe just "all the deletions have been upstreamed") in the PR description

File: Mathlib/Data/Nat/Digits.lean
Comment (eric-wieser) [inline Mathlib/Data/Nat/Digits.lean:175]: Can you explain what happened here in the PR description?

**Review by eric-wieser — APPROVED** [review]
bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**
